### PR TITLE
Add a new translate command.

### PR DESCRIPTION
**Pull request by anotherkai**
This pull request introduces a new `translate` command to the moreCommands plugin. It uses mymemory's (by translated labs) API to retrieve text and either send to the current channel or "preview" the result. There is currently a 50k character limit per day on this due to the API being free.

### CODE REVIEW

**Comment by MrDiamondDog** [comment]
This should really be apart of the translate plugin, not this one

**Comment by anotherkai** [comment]
> This should really be apart of the translate plugin, not this one

See I would if I knew how to work with Modals, I'm only just learning how to write functional plugins so for the meanwhile, I will be trying to get this added the moreCommands plugin.

File: src/plugins/moreCommands/index.ts [hunk 1]
```diff
@@ -18,10 +18,43 @@
...
+        } else {
+            throw new Error('Translation failed');
+        }
+    } catch (error) {
+        throw new Error(`Error: ${error.message}`);
+    }
+};
 
-function mock(input: string): string {
+function mock(input) {
```
Comment (MrDiamondDog) [inline src/plugins/moreCommands/index.ts:57]: why

File: src/plugins/moreCommands/index.ts [hunk 2]
```diff
@@ -18,10 +18,43 @@
 
 import { ApplicationCommandInputType, findOption, OptionalMessageOption, RequiredMessageOption, sendBotMessage } from "@api/Commands";
 import { Devs } from "@utils/constants";
-import definePlugin from "@utils/types";
+import { definePluginSettings } from "@api/Settings";
+import definePlugin, { definePluginSettings, OptionType } from "@utils/types";
```
Comment (MrDiamondDog) [inline src/plugins/moreCommands/index.ts:22]: ?

File: src/plugins/moreCommands/index.ts [hunk 3]
```diff
@@ -18,10 +18,43 @@
...
 
+const settings = definePluginSettings({
+    show_match: {
+        type: OptionType.BOOLEAN,
+        description: "Shows a match number (like 0.85) to other sources.",
+        default: false,
+        restartNeeded: false,
+    },
```
Comment (MrDiamondDog) [inline src/plugins/moreCommands/index.ts:30]: What does this option do? the description doesn't make sense

**Review by MrDiamondDog — CHANGES_REQUESTED** [review]
The reason I suggested it be moved is because then you can use the same translating api as the translate plugin, which uses google translate and thus does not have a quota

File: src/plugins/moreCommands/index.ts (hunk 1 above)
Comment (anotherkai) [reply src/plugins/moreCommands/index.ts:57]: I actually didn't know I did this :sob:

**Comment by Vendicated** [comment]
yeah u should add this as a command to the Translate plugin instead, it already has a translate function so you just need to add the command

File: src/plugins/moreCommands/index.ts [hunk 4]
```diff
@@ -65,9 +57,8 @@ function mock(input) {
 export default definePlugin({
     name: "MoreCommands",
     description: "echo, lenny, mock, translate",
-    authors: [Devs.Arjix, Devs.echo, Devs.Samu],
+    authors: [Devs.Arjix, Devs.echo, Devs.Samu, {name: "Kai :3", id: 1205727693811879997n}],
```
Comment (Scyye) [inline src/plugins/moreCommands/index.ts:60]: Add yourself to the devs constant

File: src/plugins/moreCommands/index.ts [hunk 5]
```diff
@@ -54,7 +46,7 @@ const translateText = async (sourceLang, targetLang, text) => {
     }
 };
 
-function mock(input) {
+function mock(input: string): string {
```
Comment (Scyye) [inline src/plugins/moreCommands/index.ts:49]: I- why??

**Review by Scyye — CHANGES_REQUESTED** [review]
This should 100% not be in moreCommands. Just move it over to the translation plugin.

File: src/plugins/moreCommands/index.ts (hunk 5 above)
Comment (anotherkai) [reply src/plugins/moreCommands/index.ts:49]: Did I seriously forget to symbol link :-:

Comment (anotherkai) [reply src/plugins/moreCommands/index.ts:49]: i did :facepalm:

**Comment by anotherkai** [comment]
Closed as I will be moving it over to translate plugin, will make a new pr soon.

**Pull request closed without merging**